### DISCUSSION TITLE
Generating tiles using QGIS

### DISCUSSION THREAD

**Discussion by barteis** [osmandapp/OsmAnd#11825] [Q&A] [answered]
(forked from Sqlite map from mapc2mapc - not working)

@mariush444 Regarding the three lines of GDAL code you suggested for creating Osmond-compliant tiles in QGIS....

Can this method work entirely within QGIS with GDAL installed?  On a non-Linux machine? If so, could you please walk through the steps--at least once the map is georeferenced.  Thanks!

**Comment by thompehw** [comment, marked as answer]
Using sequences of GDAL command line instructions (using Linux scripts or whatever) can make things faster if you need to do a similar operation many times. But AFAIK, they're not doing anything that you can't to in QGIS with the normal menus and plugins, in Windows or Linux.  QGIS uses GDAL. 

After georeferencing, just use the menus at Processing>Toolbox>Raster tools>Generate XYZ tiles (Directory) to generate tiles. 

Then you have to change the file extensions of every tile from .png->.png.tile or .jpg->.jpg.tile

@mariush444 has noted that OsmAnd can't handle jpg packaged into a single sqlite file, but jpg allows for much smaller files if you just place the whole directory hierarchy (I think some call it a "pyramid") into Osmand's "tiles" folder.

**Reply by barteis** [reply]
Thanks @thompehw 

It's striking that QGIS can export to 49 raster formats, none of which seems compatible with OsmAnd/Locus.  I'm curious how it came to be that 50+ raster formats even exist.  They're almost as numerous as programming languages.

**Reply by mariush444** [reply]
When you write about tiles in directory - it's not so bad.
There are to methods/system generally.
Zoom x y (used by google, osmand ...) and TMS. 
```
Zoom - 1st level of directory
x - 2nd level - sub directory
y - name of the file (y.png)

```
TMS is simillar to zxy but the y is calculate in "oposite" direction.

Here you can see how it is look like (map with the tiles)
https://www.maptiler.com/google-maps-coordinates-tile-bounds-projection/

**Comment by mariush444** [comment]
_sorry for no response. I was on small holidays._

I don't use windows but @thompehw is right, generally. There shouldn't be deference between Win&Linux.
QGIS it is not user friendly but you always find what you are looking for. 
So...
When you have your geofile (pdf or tif) and you can see in right place on the map then use menu mentioned by thom or look at the screenshot below or in the previous topic.
Choose and Execute from menu Toolbox->Raster tools->Generate XYZ tiles (Directory)
Then you have to choose area that should be converted to tiles.
From the menu (in red) choose "drawn on canvas". Other options don't work for me or do what I don't expect. 
Set the min&max zoom as you like/want.
Set the destination directory and press Run.

Do not forget to rename all generated tiles from .png -> .png.tiles (It is must be done for OSMAND)

![image](https://user-images.githubusercontent.com/66887280/120975026-1572a980-c771-11eb-8efe-e9af30e3b65b.png)

Let me know if you need more help.

PS
Can you install M2M on Windows and verify if you can run commands
`gdalwarp gdal_translate gdal2tiles.py `
in cmd.exe?

**Reply by barteis** [reply]
Thanks @mariush444 -- I was able to do this using your instructions for QGIS.  Then on Windows, renaming the files from *.png to *.png.tile using the CMD line:
forfiles /S /M *.png /C "cmd /c rename @file @fname.png.tile"

I'm thinking this is at the outer limits of what I'd want create a tutorial video for. (As in [these](https://sites.google.com/view/bartetutorials/home).) Especially as the batch file renaming varies by operating system.  

Re: your PS: I have M2M installed, opened it, and tried to run the command on Windows cmd (yes?).  It returns "'gdalwarp' is not recognized as an internal or external command operable program or batch file."
